diff --git a/src/lib_json/json_reader.cpp b/src/lib_json/json_reader.cpp
index 896bf1b..6624cb0 100644
--- a/src/lib_json/json_reader.cpp
+++ b/src/lib_json/json_reader.cpp
@@ -1654,6 +1654,12 @@ bool OurReader::decodeDouble(Token& token, Value& decoded) {
   const String buffer(token.start_, token.end_);
   IStringStream is(buffer);
   if (!(is >> value)) {
+    // the value could be lower than numeric_limits<double>::min(), in this situtation we should return the value with the gurantee
+    // of conversion which has been performed and no occurances of range error.
+    if ((value > 0 && value < std::numeric_limits<double>::min()) || (value < 0 && value > -std::numeric_limits<double>::min())) {
+      decoded = value;
+      return true;
+    }
     if (value == std::numeric_limits<double>::max())
       value = std::numeric_limits<double>::infinity();
     else if (value == std::numeric_limits<double>::lowest())

